cmake/CMakeLists.txt

```diff
@@ -613,6 +613,7 @@ if (onnxruntime_USE_TENSORRT)
   if (WIN32)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /DELAYLOAD:nvinfer.dll")
   endif()
+  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
```

Comment:
why do we need this?
if needed, add it to specific module vs applying globally.